src/PersistStore.ts

```diff
@@ -105,6 +107,15 @@ export class PersistStore<T, P extends keyof T> {
             ].every(Boolean);
 
             if (allowPropertyHydration) {
+              if (target[propertyName] instanceof ObservableMap) {
+                const content = data[propertyName];
+
+                if (isArrayForMap(content)) {
+                  target[propertyName] = new Map(content);
+                  return;
+                }
+              }
+
               target[propertyName] = data[propertyName];
```

Comment:
```suggestion
              const propertyData = data[propertyName];
              
              if (propertyData instanceof ObservableMap && isArrayForMap(propertyData)) {
                target[propertyName] = new Map(propertyData);
              } else {
                target[propertyName] = propertyData;
              }
```

Reply:
You mistake here.

`target[propertyName] instanceof ObservableMap` is not equal `properyData instanceof ObservableMap`

Reply:
your right, I updated the example above